seed/views/v3/building_files.py

```diff
@@ -0,0 +1,174 @@
+# !/usr/bin/env python
+# encoding: utf-8
+"""
+:copyright (c) 2014 - 2020, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+import os
+import zipfile
+from tempfile import NamedTemporaryFile
+from django.core.files.uploadedfile import InMemoryUploadedFile
+from django.http import JsonResponse
+from django.utils.decorators import method_decorator
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.parsers import MultiPartParser
+
+from seed.lib.superperms.orgs.decorators import has_perm_class
+from seed.models import BuildingFile, Cycle
+from seed.serializers.building_file import BuildingFileSerializer
+from seed.serializers.properties import PropertyViewAsStateSerializer
+from seed.utils.api_schema import AutoSchemaHelper, swagger_auto_schema_org_query_param
+from seed.utils.viewsets import SEEDOrgReadOnlyModelViewSet
+
+
+@method_decorator(swagger_auto_schema_org_query_param, name='list')
+@method_decorator(swagger_auto_schema_org_query_param, name='retrieve')
+class BuildingFileViewSet(SEEDOrgReadOnlyModelViewSet):
+    model = BuildingFile
+    orgfilter = 'property_state__organization'
+    parser_classes = (MultiPartParser,)
+
+    def get_serializer_class(self):
+        # Super hacky, but this allows us to have separate serializers for different
+        # views
+        # This is necessary b/c bad swagger docs were being generated due to the
+        # serializer for the `create` view
+        if self.request.method.lower() == 'post':
+            return None
+        return BuildingFileSerializer
```

Comment:
I'm not happy with this, but it was better than the other options:
- apply serializer to entire class, then have some unwanted params in the swagger docs for the create view
- apply serializer on swagger_auto_schema_decorator for the list and retrieve endpoints. This seemed less preferable b/c then DRF wouldn't know about the serializer and use it.

Reply:
I think this isn't bad/hacky. Actually, according to the DRF docs, this sounds exactly like what overriding the get_serializer_class was meant for:  

>May be overridden to provide dynamic behavior, such as using different serializers for read and write operations...
https://www.django-rest-framework.org/api-guide/generic-views/#get_serializer_classself

I'd vote to remove all or most of those comments actually.